binr/ragg2/ragg2.c

```diff
@@ -108,6 +109,257 @@ static int openfile (const char *f, int x) {
+		use_clang = true;
+		strcpy (TEXT, "0.__TEXT.__text");
+	} else if (!strcmp (TRIPLET, "darwin-arm-32")) {
+		strcpy (CC, "xcrun --sdk iphoneos gcc -arch armv7 -miphoneos-version-min=0.0");
+		use_clang = true;
+		strcpy (TEXT, "0.__TEXT.__text");
+	}
+
+	snprintf (buffer, sizeof (buffer), "%s -nostdinc -include %s/%s/sflib.h",
+	  			CFLAGS, SFLIBPATH, TRIPLET);
+	strncpy (CFLAGS, buffer, sizeof (CFLAGS) - 1);
+
+	if (use_clang) {
+		snprintf (buffer, sizeof (buffer), "%s -fomit-frame-pointer"
+		  				" -fno-zero-initialized-in-bss", CFLAGS);
+		strncpy (CFLAGS, buffer, sizeof (CFLAGS) - 1);
+	} else { 
+		snprintf (buffer, sizeof (buffer), "%s -z execstack -fomit-frame-pointer"
+		  				" -finline-functions -fno-zero-initialized-in-bss", CFLAGS);
+		strncpy (CFLAGS, buffer, sizeof (CFLAGS) - 1);
+	}
+	snprintf (buffer, sizeof (buffer), "%s -nostdlib", LDFLAGS);
+	strncpy (LDFLAGS, buffer, sizeof (LDFLAGS) - 1);
+
+	printf ("==> Compile\n");
+	printf ("%s %s -o %s.tmp -S -Os %s\n", CC, CFLAGS, file, file);
+
+	free (output);
+	output = r_sys_cmd_strf ("(%s %s -o %s.tmp -S -Os %s) 2>&1", CC, CFLAGS, file, file);
+	if (!output) {
+		eprintf ("Compilation failed!\n");
+		goto fail;
+	}
+	printf ("%s", output);
+
+	if ((fileExt = malloc (strlen (file) + 6)) == NULL) {
+		eprint ("Malloc for fileExt failed!\n");
+		goto fail;
+	}
+
+	sprintf (fileExt, "%s.s", file);
+
+	fileExtFd = open (fileExt, O_CREAT | O_RDWR, 0644);
+	if (fileExtFd == -1) {
+		eprintf ("Error while opening %s.s\n", file);
+		goto fail;
+	}
+	write (fileExtFd, SHDR, strlen (SHDR));
+	close (fileExtFd);
+
+	free (output);
+	output = r_sys_cmd_strf ("cat %s.tmp | sed -e s,rdata,text, -e 's,get_pc_thunk.bx,__getesp__,g' "
+	  				 "| grep -v .cstring | grep -v size | grep -v ___main | grep -v section "
+					 "| grep -v __alloca | grep -v zero | grep -v cfi >> %s.s", file, file);
+	if (!output) {
+		eprintf ("Write in %s.s failed!\n", file);
+		goto fail;
+	}
+	printf ("%s", output);
+
+	printf ("==========================\n");
+	
+	printf ("==> Assemble\n");
+	printf ("%s %s -Os -o %s.o %s.s\n", CC, LDFLAGS, file, file);
+
+	free (output);
+	output = r_sys_cmd_strf ("%s %s -Os -o %s.o %s.s", CC, LDFLAGS, file, file);
+	if (!output) {
+		eprintf ("Assembly failed!\n");
+		goto fail;
+	}
+	printf ("%s", output);
+
+	printf ("==> Link\n");
+	printf ("rabin2 -o %s.text -O d/S/%s %s.o\n", file, TEXT, file);
+
+	free (output);
+	output = r_sys_cmd_strf ("rabin2 -o %s.text -O d/S/%s %s.o", file, TEXT, file);
+	if (!output) {
+		eprintf ("Linkage failed!\n");
+		goto fail;
+	}
+
+	sprintf (fileExt, "%s.o", file);
+	if (access (fileExt, F_OK) == -1) {
+		eprintf ("Cannot find %s.o\n", file);
+		goto fail;
+	}
+
+	free (output);
+	output = r_sys_cmd_strf ("du %s.text | awk '{print $1}'", file);
+	if (!output) {
+		eprintf ("Failed at test 'du %s.text'\n", file);
+		goto fail;
+	}
+
+	if (!strcmp (output, "0")) {
```

Comment:
Uh? 0?

Reply:
it is disk usage so it may be 0 ? (I kinda copied/pasted the code from bash)